volatility/framework/automagic/linux.py

```diff
@@ -293,6 +294,56 @@ def virtual_to_physical_address(cls, addr: int) -> int:
             return addr - 0xffffffff80000000
         return addr - 0xc0000000
 
+    @classmethod
+    def mask_mods_list(cls, context: interfaces.context.ContextInterface, layer_name: str,
+                       mods: Iterator[Any]) -> Iterator[Any]:
+        """
+        A helper function to mask the starting and end address of kernel modules
+        """
+        mask = context.layers[layer_name].address_mask
+
+        return [(utility.array_to_string(mod.name), mod.get_module_base() & mask, (mod.get_module_base() & mask) + mod.get_core_size())
+                for mod in mods]
+
+    @classmethod
+    def generate_kernel_handler_info(
+            cls,
+            context: interfaces.context.ContextInterface,
+            layer_name: str,
+            kernel,  # ikelos - how to type this??
+            mods_list: Iterator[Any]):          
+
+        mask = context.layers[layer_name].address_mask
+
+        start_addr = kernel.object_from_symbol("_text")
+        start_addr = start_addr.vol.offset & mask
+
+        end_addr = kernel.object_from_symbol("_etext")
+        end_addr = end_addr.vol.offset & mask
+
+        return [("__kernel__", start_addr, end_addr)] + \
+            LinuxUtilities.mask_mods_list(context, layer_name, mods_list)
+
+    @classmethod
+    def lookup_module_address(cls, context: interfaces.context.ContextInterface, handlers: Iterator[Any],
+                              target_address):
+        mod_name = "UNKNOWN"
+        symbol_name = "N/A"
+
+        for name, start, end in handlers:
+            if start <= target_address <= end:
+                mod_name = name
+                if name == "__kernel__":
+                    symbols = list(
+                        context.symbol_space.get_symbols_by_location(target_address))
+
+                    if len(symbols) > 0:
+                        symbol_name = str(symbols[0].split(constants.BANG)[1]) if constants.BANG in symbols[0] else \
```

Comment:
There's a lot of casting things as strings in here, are the not already strings?

Reply:
It looks like they are. This is from the automagic/mac file if you want to change it on that side as well

Reply:
Yeah, I would, sorry, always tidying things up...  5;)